hudi-common/src/main/java/org/apache/hudi/common/table/log/HoodieMergedLogRecordScanner.java

```diff
@@ -108,30 +112,85 @@ protected HoodieMergedLogRecordScanner(FileSystem fs, String basePath, List<Stri
     }
   }
 
-  protected void performScan() {
+  /**
+   * Scans delta-log files processing blocks
+   */
+  public final void scan() {
+    scan(false);
+  }
+
+  public final void scan(boolean skipProcessingBlocks) {
+    scanInternal(Option.empty(), skipProcessingBlocks);
+  }
+
+  /**
+   * Provides incremental scanning capability where only provided keys will be looked
+   * up in the delta-log files, scanned and subsequently materialized into the internal
+   * cache
+   *
+   * @param keys to be looked up
+   */
+  public void scanByFullKeys(List<String> keys) {
```

Comment:
this method naming is confusing w/ scan(), scanInternal(). may be we should rename this to scanAndGetRecords(). bcoz, for both scanByFullKeys and scanByKeyPrefixes, immediately caller calls logRecordScanner.getRecords(). 

This will differentiate from one time preprocessing using scan or scanInternal(). 
I am even open to renaming scan() to something like load/fetchRecordsFromStorage.

Reply:
Yeah, good point. I've kept it consistent w/ `scan`, b/c semantic of these methods is the same and they differ just in scope.

I'm open to it, but need to think more if it's a good idea to rename the `scan` methods as these could be relied upon (we're not annotating it as API, but it is an API of the Reader)